packages/components/avatar/src/use-avatar.ts

```diff
@@ -199,12 +201,17 @@ export function useAvatar(originalProps: UseAvatarProps = {}) {
     (props = {}) => ({
       ref: imgRef,
       src: src,
-      disableAnimation,
       "data-loaded": dataAttr(isImgLoaded),
       className: slots.img({class: classNames?.img}),
-      ...mergeProps(imgProps, props),
+      ...mergeProps(
+        imgProps,
+        props,
+        filterDOMProps({disableAnimation} as DOMAttributes<DOMElement>, {
+          enabled: shouldFilterDOMProps,
+        }),
+      ),
     }),
-    [slots, isImgLoaded, imgProps, disableAnimation, src, imgRef],
+    [slots, isImgLoaded, imgProps, disableAnimation, src, imgRef, filterDOMProps],
```

Comment:
```suggestion
    [slots, isImgLoaded, imgProps, disableAnimation, src, imgRef, filterDOMProps],
```

@wingkwong as `filterDOMProps` is a function so it will cause the useCallback act as a normal function